entrypoint.sh

```diff
@@ -133,7 +133,7 @@ fi
 
 
 if [[ ${_RUN_AS_ROOT} == "true" ]]; then
-  [[ $(id -u) -eq 0 ]] && ( "$@" ) || ( echo "ERROR: RUN_AS_ROOT env var is set to true but the user has been overriden and is not running as root"; exit 1 )
```

Comment:
what this translates to:

```
if _RUN_AS_ROOT is true: (This defaults to true, the only way not be true is to give it an explicit value)
    if id -u is 0 (root):
      run CMD
    else
      error and exit 1
```

This is correct. It means: If run as root set and true AND root: run it. else error
The only way this can fail is if RUN_AS_ROOT is anything other than `true` specifically and id -u is not 0. this would mean that the user running it is not in fact root

I would build your own version of this that spits out what `id -u` is here as well as the value of `_RUN_AS_ROOT` because my guess is that its `true` and not root

if RUN_AS_ROOT is anything other than specifically `true` you wouldnt end up here. 

> I managed to work around it by setting the env var RUN_AS_ROOT to an empty string
this would cause it to default to `true` no matter what

```
✗ export RUN_AS_ROOT=""
✗ _RUN_AS_ROOT=${RUN_AS_ROOT:="true"}
✗ echo $_RUN_AS_ROOT
true
```

Reply:
In my case the user is root, I hadn't set RUN_AS_ROOT, but I hit this error, maybe because of https://github.com/koalaman/shellcheck/wiki/SC2015 ?

Reply:
Shell check may flag it but that's not the case here. I can't reproduce this

Reply:
I did some more testing this morning, I made the following change to a running container, in entrypoint.sh:

```
  [[ $(id -u) -eq 0 ]] && ( "$@" ) || ( echo "ERROR: RUN_AS_ROOT env var is set to true but the user has been overriden and is not running as root"; exit 1 )
```

To become:

```
  [[ $(id -u) -eq 0 ]] && ( "$@" ) || ( echo "ERROR: RUN_AS_ROOT env var is set to true but the user has been overriden and is not running as root.  Current user configured as $(whoami)"; exit 1 )
```

This change just prints out the username of the current user, then I started the container with:

```
/usr/bin/docker run \
  --rm \
  --name="github-runner-${REPO}-$1" \
  --privileged \
  --dns 192.168.200.1 \
  --network=host \
  -e REPO_URL="https://github.com/intenthq/${REPO}" \
  -e ACCESS_TOKEN="${PAT_TOKEN}" \
  -e RUNNER_NAME="teamcity-${REPO}-$1" \
  -e RUNNER_WORKDIR="/tmp/github-runner/work_${REPO}-$1" \
  -u root \
  -v /var/run/docker.sock:/var/run/docker.sock \
  -v /tmp/github-runner:/tmp/github-runner \
  -v /dev/log:/dev/log \
  -e LABELS="${LABELS}" \
   rumbles/github-runner:testing
```

And I got this in the logs:


```
Aug 30 07:57:37 build-1 github-runner.sh[2128657]: ERROR: RUN_AS_ROOT env var is set to true but the user has been overriden and is not running as root. Current user configured as root
```

I think with the current logic, if either of these fail:

```
 [[ $(id -u) -eq 0 ]] && ( "$@" )
```

The error state can run, which I don't think is what is wanted, it should only run if RUN_AS_ROOT is true but the user is not root.

My initial PR had made the wrong assumption, I will update it